### ISSUE TITLE
Teach `breadboard debug` to handle imports in the board TS files.

### ISSUE THREAD

**Issue by dglazkov** [breadboard-ai/breadboard#729]
As I've been playing with `breadboard debug` I am starting to wonder: should it deal with TypeScript compilation at all? It adds a ton of complexity to code and adds strange side effects (like, you can't import another TS file into it... something's off with paths?), but it's also super-trivial to implement as part of the initial project (`tsx make-graphs.ts` or something like that).

What if instead, we focused on only serving and looking for `json` files? My guess is that it would make the logic simpler and reduce complexity of the breadboard CLI, while also give devs more room on how to compile JSON (like, from Python, etc.)

@PaulKinlan WDYT?

**Comment by dglazkov** [comment]
Ah, it looks like the problem is that if we want to do `--watch`, we also want to launch something on changes to the `ts` files. Darn it.

**Comment by PaulKinlan** [comment]
What do you mean you can't import another typescript file? Can you share an example?

I think breadboard as a whole should have a shared compilation library (the CLI has exports to it's functionality) that can be used anywhere and is consistent. i.e, if we offer the UI to build TS then we'd need it etc etc.

In addition, we have `breadboard make` where there should be similar logic to `make-graphs` - I'd really like to remove this file across our in-branch projects and replace it with the exported logic in `breadboard make`. 

Broadly my pov is that we need to have a simple tool for out-of-branch developers to compile their boards. CLI seems to be the most sensible as an interface, I'm ambivalent to where the logic lives.

If we make the `breadboard debug` only serve JSON, we still don't help with TS developers accidentally pointing to a TS board instead of a JSON board. Some options:

* Teach the `invoke` how to compile, so who cares if you link to a TS file or not. Invoke fetches the file, and determines the correct compile and returns a board (this way we only compile the board being linked at runtime which has higher latency but you don't need a pre-scanner)
* Have a pre-scanner at build time that will look for anything it can statically determine is an invoke-able and compile it. It not currently possible to compile ahead of time and know for certain that the node is an "invoke" from core-kit (so we'd need to fix kits and namespaces properly imo)
* Do nothing other than tell people they need to link to JSON.

**Comment by dglazkov** [comment]
I the `import` doesn't work because of:

https://esbuild.github.io/api/#transform
> This is a limited special-case of [build](https://esbuild.github.io/api/#build) that transforms a string of code representing an in-memory file in an isolated environment that's completely disconnected from any other files.

**Comment by dglazkov** [comment]
Renaming the issue to make it more focused.

**Comment by dglazkov** [comment]
> Broadly my pov is that we need to have a simple tool for out-of-branch developers to compile their boards.

The thing that I am observing is that most of the time, I want a TS environment where I write programs that eventually compile to boards. Especially when building light kits, I want to reuse type definitions, organize them, etc. -- and only when I say "compile" do I want these things to become boards.

This tells me that a simple CLI idea has an unbounded scope -- it will eventually grow to accommodate all the things that are part of TypeScript development experience. Does that make sense?

**Comment by PaulKinlan** [comment]
Got it.

I'll have a look at how you are thinking of light-kits. When I first did this debug tool, there wasn't a lot of places that we imported anything other than breadboard and the kits... 

We probably just want to do what @paullewis did and import the typescript compiler and not the esbuild tooling if we want to accomodate all of typescript. It's a pretty heavy weight dependency, but that might be ok.

I can work with this. Short-term, look at fixing the immediate issue. Longer-term abstracting the functionality so all parts of breadboard can use the functionality.
